devices/sam/samd21.xml

```diff
@@ -0,0 +1,704 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<modm version="0.4.0">
+  <!-- WARNING: This file is generated by the modm device file generator. Do not edit! -->
+  <device platform="sam" family="d" series="21" pin="e|g|j" flash="15|16|17|18" variant="a|au|b|bu|cu|d|du|l">
+    <naming-schema>{platform}{family}{series}{pin}{flash}{variant}</naming-schema>
```

Comment:
I was having a lot of issues with package. it is only specified if it matters and the only thing it changes is the pinout. For the three chips I am working with now only the `u` package is unique. There is a list of compatible variants at the beginning of each `atsd` file, but then there is a lot of duplication. So I just made the variant include the package, so the identifier now has a 1 or 2 character variant. This makes it match the way microchip names their files as well.

Initally I had all packages that didn't specify the package default to `m`. but that might be confusing for someone looking for the `a` package. I am also not sure it will scale. I don't use other SAM chips but there might be other chips where m is the weird one and u is the default.

Reply:
Hm, I'd still make it a separate *key* (with an empty string for no package like `package="|u"`). The reason is that then you don't drag this hack into the Jinja templates as well, because the identifier is used to split up the `modm:target` string into it's components.
```
%% if target.variant in ['a', 'au', 'b', 'bu']
    enum class VariantABonly {};
%% endif
```
Instead of:
```
%% if target.variant in ['a', 'b']
    enum class VariantABonly {};
%% endif
```

But lets leave this for some later iteration, since this may not even be that important.